src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -402,6 +402,20 @@ def remove_additional_prop_layer(obj, converted_dic):
                 converted_dic.update(converted_dic.pop('additionalProperties'))
         return converted_dic
 
+    @staticmethod
+    # copy params so that original params remain unmodified when.
+    # params need to be unmodified in the event that a command is executed multiple times against different --ids
+    def _copy_params(params):
+        import copy
+        new_params = {}
+        for key, value in params.items():
+            # TODO: is try except more robust here???
```

Comment:
I think being explicit here is more self-documenting that try/except. Just my opinion though.